WMFData/Sources/WMFData/Data Controllers/Year In Review/WMFYearInReviewDataController.swift

```diff
@@ -815,53 +545,46 @@ import CoreData
         service.perform(request: request, completion: completionHandler)
     }
 
-    private func populateDonatingSlide(report: CDYearInReviewReport, backgroundContext: NSManagedObjectContext) throws {
-
-        guard let iosFeatureConfig = developerSettingsDataController.loadFeatureConfig()?.ios.first,
-              let yirConfig = iosFeatureConfig.yir(yearID: targetConfigYearID) else {
-            return
-        }
+    public func createNewYearInReviewReport(year: Int, slides: [WMFYearInReviewSlide]) async throws {
+        let newReport = WMFYearInReviewReport(year: year, slides: slides)
 
-        guard let dataPopulationStartDate = yirConfig.dataPopulationStartDate,
-              let dataPopulationEndDate = yirConfig.dataPopulationEndDate else {
-            return
-        }
+        try await saveYearInReviewReport(newReport)
+    }
 
-        let donateHistory = WMFDonateDataController.shared.loadLocalDonationHistory(startDate: dataPopulationStartDate, endDate: dataPopulationEndDate)
-        let donateCount = donateHistory?.count ?? 0
+    public func fetchYearInReviewReport(forYear year: Int) throws -> WMFYearInReviewReport? {
+        assert(Thread.isMainThread, "This report must be called from the main thread in order to keep it synchronous")
```

Comment:
Optional: In the spirit of making things easier for the Swift 6 migration, we could evaluate if this class should be a main actor.
But only if you feel like exploring Swift concurrency a little, since this code works fine.